code/modules/economy/holopay.dm

```diff
@@ -134,15 +127,18 @@
 			return TRUE
 	return FALSE
 
-/obj/machinery/holopay/attackby(obj/item/held_item, mob/user, params)
+/obj/effect/overlay/holopay/attackby(obj/item/held_item, mob/user, params)
+	var/obj/item/card/id/linked_card = card_ref.resolve()
+	if(QDELETED(linked_card) || !istype(linked_card, /obj/item/card/id))
```

Comment:
You don't actually need this qdeleted check, just a check for existence. weakrefs won't return the ref if the object is being destroyed